### PR TITLE
add MySQL integer types

### PR DESCRIPTION
**Pull request by b264**
For use in a factory to match model validations.  Includes edge cases for testing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,20 @@ FactoryHelper::String.random([0, 6]) #=> "I轤𣴒P溟L"
 ```
 
 
+###FactoryHelper::MySQL
+-----------------
+Generates values for a factory per the MySQL datatypes.  Accepts options unsigned, min, max.
+
+```ruby
+FactoryHelper::MySQL.integer #=> 1729
```
Comment (NikitaAvvakumov) [inline README.md]: Include a note that `.integer` & `.int` are equivalent?

File: lib/factory-helper/my_sql.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      def int_with_edge_cases opt= {:min => -128, :max => 127}
+        [ [-1, 1, 0, opt[:min], opt[:max]].keep_if do |element|
+            (opt[:min].. opt[:max]).include?(element)
+          end.sample(:random => FactoryHelper::Config.random),
+          FactoryHelper::Config.random.rand(opt[:min].. opt[:max])
+        ].compact.sample(:random => FactoryHelper::Config.random)
+      end
+
+      def int_sanitize opt
+        check_bounds opt.tap { raise 'Argument list must be a hash.' unless opt.kind_of? Hash
```
Comment (NikitaAvvakumov) [inline lib/factory-helper/my_sql.rb]: Why is the hash validation handled differently from the others? And why is `check_bounds` extracted into a separate method?

What do you think about extracting the conditionals into individual one-line methods? `if min_not_an_integer`, `if min_greater_than_max` etc.

**Comment by NikitaAvvakumov** [comment]
So, I don't want to sound negative, but... (great intro, eh?)

My biggest gripe is that I find the method definitions difficult to follow. A multi-line `if` statement as an array element stretches the syntax boundary too far, in my opinion. The `int_with_edge_cases` method really takes this to another level :smile: I would also argue that while Sandi's 5-line rule is aimed at enforcing the SRP, these methods break that principle despite fitting within the limit. One of those "letter of the law" vs. "spirit of the law" things...

Given the "nesting" of smaller-number methods within the larger-number ones, I can't help but think that there is a way to handle this in a single recursive method. Unfortunately, recursion makes my head hurt, so I am unlikely to actually come up with a solution. Still, I feel that there is something to explore here, given that all methods are almost identical and what changes are the min & max limits and the smaller-number method they call.

Without that significant reworking, and I don't have the confidence that I'll get anywhere with it, there still seem to be ways of implementing the current methods clearer. What do you think about the following:

``` ruby
LIMITS = {
  :smallint => {
    true => { :min => 0, :max => 65535 },
    false => { :min => -32768, :max => 32767 }
  }
}

def smallint(opt = {})
  limits = LIMITS[:smallint][!!opt[:unsigned]].merge(opt)
  smallint = int_with_edge_cases(limits)
  seeded_sample(tinyint(opt), smallint)
end

private

  def seeded_sample(*args)
    args.sample(:random => FactoryHelper::Config.random)
  end
```

:arrow_up: leaves specs green

Now, could we extract the name/symbol `smallint` from the above method and pass it in as an argument from `MySQL.method_missing`? And have all method names in an array so that `smallint` knows to call `tinyint`, which is at `i-1`? (And, of course, there is no `tinyint`, just `method_missing` again...) I'll keep thinking. Does this sound like it's worth the effort?

File: README.md (hunk 1 above)
Comment (b264) [inline README.md]: This is part of the MySQL specs but we can add it for clarity.  I was deciding whether to include the note or not.

**Comment by b264** [comment]
I don't like `method_missing` because the class may be open to non-integer types in the future and also I'm not sure if that would bypass the determinism spec.  Although we should look into that in a separate PR

File: lib/factory-helper/my_sql.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
+module FactoryHelper
+  # created for MySQL 5.0 https://dev.mysql.com/doc/refman/5.0/en/numeric-types.html
+  class MySQL < Base
+    INTEGERS= {
+      :tinyint => {
+        false => {:min => -128, :max => 127,},
```
Comment (NikitaAvvakumov) [inline lib/factory-helper/my_sql.rb]: Extra commas in hashes

**Comment by NikitaAvvakumov** [comment]
Looks good to me. Ready? :ship:

**Pull request merged**